FileShareAdminClient/FileShareApiAdminClient.cs

```diff
@@ -369,61 +372,63 @@ private async Task<IResult<AddFileToBatchResponse>> AddFiles(IBatchHandle batchH
                 }
                 else
                 {
-                    var md5Hash = stream.CalculateMD5();
-                    stream.Seek(0, SeekOrigin.Begin);
-
                     var fileBlocks = new List<string>();
                     var fileBlockId = 0;
                     var expectedTotalBlockCount = (int)Math.Ceiling(stream.Length / (double)maxFileBlockSize);
                     progressUpdate((0, expectedTotalBlockCount));
 
                     var buffer = new byte[maxFileBlockSize];
-                    while (true)
+
+                    using (var md5 = MD5.Create())
+                    using (var cryptoStream = new CryptoStream(stream, md5, CryptoStreamMode.Read))
                     {
-                        fileBlockId++;
-                        var ms = new MemoryStream();
+                        while (true)
+                        {
+                            fileBlockId++;
+                            var ms = new MemoryStream();
 
-                        var read = stream.Read(buffer, 0, maxFileBlockSize);
-                        if (read <= 0) break;
-                        ms.Write(buffer, 0, read);
+                            var read = cryptoStream.Read(buffer, 0, maxFileBlockSize);
+                            if (read <= 0) break;
+                            ms.Write(buffer, 0, read);
 
-                        var fileBlockIdAsString = fileBlockId.ToString("D5");
-                        var putFileUri = $"batch/{batchHandle.BatchId}/files/{fileName}/{fileBlockIdAsString}";
-                        fileBlocks.Add(fileBlockIdAsString);
-                        ms.Seek(0, SeekOrigin.Begin);
+                            var fileBlockIdAsString = fileBlockId.ToString("D5");
+                            var putFileUri = $"batch/{batchHandle.BatchId}/files/{fileName}/{fileBlockIdAsString}";
+                            fileBlocks.Add(fileBlockIdAsString);
+                            ms.Seek(0, SeekOrigin.Begin);
 
-                        var blockMD5 = ms.CalculateMD5();
+                            var blockMD5 = ms.CalculateMD5();
```

Comment:
Not sure this is correct, looks like you're still using the memory stream to calculate the MD5 hash?

Reply:
Sorry, I take this back, it's calculating the block's MD5.